moabb/evaluations/splitters.py

```diff
@@ -101,3 +110,175 @@ def split(self, y, metadata):
+
+    Yields
+    ------
+    train : ndarray
+        The training set indices for that split.
+
+    test : ndarray
+        The testing set indices for that split.
+    """
+
+    def __init__(
+        self,
+        cv_class: type[BaseCrossValidator] = LeaveOneGroupOut,
+        shuffle: bool = False,
+        random_state: int = None,
+        **cv_kwargs,
+    ):
+        self.cv_class = cv_class
+        self.cv_kwargs = cv_kwargs
+        self._cv_kwargs = dict(**cv_kwargs)
+
+        self.shuffle = shuffle
+        self.random_state = random_state
+
+        if not shuffle and random_state is not None:
+            raise ValueError("`random_state` should be None when `shuffle` is False")
+
+        self._rng = check_random_state(self.random_state) if self.shuffle else None
+
+        if shuffle and (
+            "shuffle" not in inspect.signature(cv_class).parameters.keys()
+            and
+            # Some cross-validation strategies use 'random_state' instead of 'shuffle'
+            # where the Shuffle is already in the name of the class
+            "random_state" not in inspect.signature(cv_class).parameters.keys()
+        ):
+            raise ValueError(
+                f"Shuffling is not supported for {cv_class.__name__}. "
+                "The `shuffle` parameter will be ignored. "
+                "Choose a different cross-validation strategy that supports shuffling."
+            )
+
+        params = inspect.signature(self.cv_class).parameters
+        for p, v in [
+            ("shuffle", shuffle),
+            ("random_state", self._rng),
+        ]:
+            if p in params:
+                self._cv_kwargs[p] = v
+
+    def get_n_splits(self, metadata):
+        """
+        Return the number of splits for the cross-validation.
+
+        The number of splits is the number of subjects times the number of splits
+        of the inner cross-validation strategy.
+
+        We try to keep the same behaviour as the sklearn cross-validation classes.
+
+        Parameters
+        ----------
+        metadata: pd.DataFrame
+            The metadata containing the subject and session information.
+
+        Returns
+        -------
+        n_splits: int
+            The number of splits for the cross-validation
+        """
+        subjects = metadata["subject"].unique()
+        n_splits = 0
+        for subject in subjects:
+            subject_metadata = metadata.query("subject == @subject")
+            sessions = subject_metadata["session"].unique()
+
+            if len(sessions) <= 1:
+                continue  # Skip subjects with only one session
+
+            # To make sure that when I shuffle the subject, I shuffle the same way
+            # the session when the object is created
+            if self.shuffle:
+                self._rng = check_random_state(self.random_state)
+                self._cv_kwargs["random_state"] = self._rng
```

Comment:
We don't need this as the number of split should not depend of this parameter.

Reply:
true!